kafka-connect-cloud-common/src/main/scala/io/lenses/streamreactor/connect/cloud/common/source/state/ReaderManagerBuilder.scala

```diff
@@ -43,6 +44,7 @@ object ReaderManagerBuilder extends LazyLogging {
   def apply[M <: FileMetadata](
     root:                       CloudLocation,
     path:                       CloudLocation,
+    config:                     CloudSourceConfig[M],
```

Comment:
rather than passing through the whole config object it would be better here to just pass the `CompressionCodec`.  It reduces the amount that the class needs to know and makes it clearer what we are going to do with it.

(Eventually I would like to merge the `compressionCodec` into the `formatSelection`, to reduce the complexity, but it's not relevant to your change)